File: src/main/scala/com/jjw/spark/test1/checkpointTest.scala
package com.jjw.spark.test1

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object checkpointTest {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setMaster("local").setAppName("checkpoint")
    val sc = new SparkContext(conf)
    sc.setCheckpointDir("./checkpoint")
    val lines = sc.textFile("./NASA_access_log_Aug95")
    lines.checkpoint()
    lines.foreach { println}
    sc.stop()
  }
}

File: src/main/java/com/jjw/sparkCore/others/CheckPointTest.java
package com.jjw.sparkCore.others;

import java.util.Arrays;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class CheckPointTest {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf();
		conf.setMaster("local").setAppName("checkpoint");
		JavaSparkContext sc = new JavaSparkContext(conf);
		sc.setCheckpointDir("./checkpoint");
		JavaRDD<Integer> parallelize = sc.parallelize(Arrays.asList(1,2,3));
		parallelize.checkpoint();
		parallelize.count();
		sc.stop();
	
	}
}
